components/camel-sql/src/main/java/org/apache/camel/processor/idempotent/jdbc/JdbcMessageIdRepository.java

```diff
@@ -65,12 +75,12 @@ protected void doInit() throws Exception {
 
         if (tableName != null) {
             // update query strings from default table name to the new table name
-            tableExistsString = tableExistsString.replaceFirst(DEFAULT_TABLENAME, tableName);
-            createString = createString.replaceFirst(DEFAULT_TABLENAME, tableName);
-            queryString = queryString.replaceFirst(DEFAULT_TABLENAME, tableName);
-            insertString = insertString.replaceFirst(DEFAULT_TABLENAME, tableName);
-            deleteString = deleteString.replaceFirst(DEFAULT_TABLENAME, tableName);
-            clearString = clearString.replaceFirst(DEFAULT_TABLENAME, tableName);
+            tableExistsString = DEFAULT_TABLE_EXISTS_STRING.replaceFirst(DEFAULT_TABLENAME, tableName);
```

Comment:
You should use `replace` instead of `replaceFirst` to avoid using a regular expression behind the scenes